### PR TITLE
ENT-4444: Replace the usages of old  in  with the new  field.

### PR DESCRIPTION
**Pull request by saleem-latif**
**JIRA Ticket:** [ENT-4444]()

__Description:__
Create a data migration to copy the values from the `course_id` field into the `course_key` field.
Remove all references to the `course_id` field in the code, NOT removing the `course_id` field from the model in the code. change `null` back to `False` and `editable` back to `True` on the `course_key` field.

**Merge checklist:**
- [x] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `make upgrade && make requirements` have been run to regenerate requirements
- [x] `./manage.py makemigrations` has been run
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
- [x] [Version](https://github.com/edx/taxonomy-connector/blob/master/taxonomy/__init__.py) bumped
- [x] [Changelog](https://github.com/edx/taxonomy-connector/blob/master/CHANGELOG.rst) record added

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/edx/taxonomy-connector/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [GitHub Actions](https://github.com/edx/taxonomy-connector/actions), verify version has been pushed to [PyPI](https://pypi.org/project/taxonomy-connector/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [course-discovery](https://github.com/edx/course-discovery) to upgrade dependencies (including taxonomy-connector)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in course-discovery will look for the latest version in PyPi.

### CODE REVIEW

File: taxonomy/admin.py [hunk 1]
```diff
@@ -27,14 +27,14 @@ class CourseSkillsTitleAdmin(admin.ModelAdmin):
...
+    list_display = ('id', 'course_key', 'created', 'modified')
+    search_fields = ('course_key',)
 
     def save_model(self, request, obj, form, change):
         """
-        Set `course_key` to the same value as `course_id` before saving the model.
+        Set `course_id` to the same value as `course_key` before saving the model.
         """
-        obj.course_key = obj.course_id
+        obj.course_id = obj.course_key
```
Comment (sameenfatima78) [inline taxonomy/admin.py:37]: Why do we need to save the course_id now that we have discarded it?

File: test_utils/factories.py [hunk 2]
```diff
@@ -44,9 +44,10 @@ class Meta:
         """
 
         model = CourseSkills
-        django_get_or_create = ('course_id', 'skill')
+        django_get_or_create = ('course_key', 'skill')
 
     course_id = factory.LazyAttribute(lambda x: FAKER.slug())
```
Comment (sameenfatima78) [inline test_utils/factories.py:49]: should we remove this course_id field from this factory model now?

**Review by sameenfatima78 — COMMENTED** [review]
Left some comments on the PR

File: taxonomy/admin.py (hunk 1 above)
Comment (saleem-latif) [reply taxonomy/admin.py:37]: Complete details are on [this](https://openedx.atlassian.net/browse/ENT-4402?focusedCommentId=538142) comment.

In short, we can not remove the course id yet as doing so causes integrity error when creating new instances of course skills via either django admin or management command. This field will be removed in the next release.

File: test_utils/factories.py (hunk 2 above)
Comment (saleem-latif) [reply test_utils/factories.py:49]: Same as above, we can not remove as doing that will case integrity error on tests.

**Review by sameenfatima78 — APPROVED** [review]

**Pull request merged**